yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,176 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    ExtractorError,
+    int_or_none,
+    str_or_none,
+    traverse_obj,
+    urlencode_postdata,
+)
+
+
+class FreeTvBaseIE(InfoExtractor):
+    def _get_api_response(self, content_id, postdata, resource_type):
+        response = self._download_json(
+            'https://www.freetv.com/wordpress/wp-admin/admin-ajax.php',
+            content_id, data=urlencode_postdata(postdata), fatal=True,
+            note=f'Downloading {content_id} {resource_type} JSON')
+
+        if response.get('data') is False:
+            raise ExtractorError(f"Response doesn't contain {resource_type} data")
+
+        return response
+
+    def _extract_video(self, content_id, action='olyott_video_play'):
+        api_response = self._get_api_response(
+            content_id, resource_type='video',
+            postdata={
+                'action': action,
+                'contentID': content_id,
+            })
+
+        video_id = traverse_obj(api_response, ('data', 'displayMeta', 'contentID'), expected_type=str)
```

Comment:
This should be fatal
```suggestion
        video_id = api_response['data']['displayMeta']['contentID']
```